--- a/feeds/packages/net/ddns-go/Makefile
+++ b/feeds/packages/net/ddns-go/Makefile
@@ -5,12 +5,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=ddns-go
-PKG_VERSION:=6.8.1
+PKG_VERSION:=6.9.1
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://codeload.github.com/jeessy2/ddns-go/tar.gz/v$(PKG_VERSION)?
-PKG_HASH:=583eca3bb3e350c2100a660f8a3ea110ac6829d1200d3539702a81f204e3b673
+PKG_HASH:=cb15bd2a4fb48eac138920685ec3fd1088b8c8fe9e67adeea83e6ca720e83cef
 
 PKG_LICENSE:=MIT
 PKG_LICENSE_FILES:=LICENSE
