--- a/feeds/packages/lang/rust/rust-host-build.mk
+++ b/feeds/packages/lang/rust/rust-host-build.mk
@@ -44,6 +44,9 @@ define RustBinHostBuild
   define Host/Install
 	$(INSTALL_DIR) $(STAGING_DIR_HOSTPKG)/bin
 	$(INSTALL_BIN) $(HOST_INSTALL_DIR)/bin/* $(STAGING_DIR_HOSTPKG)/bin/
+
+	upx --best --lzma "$(STAGING_DIR_HOSTPKG)/bin/hbbs" || true; \
+	upx --best --lzma "$(STAGING_DIR_HOSTPKG)/bin/hbbr" || true
   endef
 endef
 
