--- a/include/target.mk
+++ b/include/target.mk
@@ -404,3 +404,36 @@ ifeq ($(TARGET_BUILD),1)
 endif
 
 endif #__target_inc
+
+OPTIMIZE_TARGETS := ddns-go \
+                    syncthing \
+                    docker \
+                    dockerd \
+                    containerd \
+                    ctr \
+                    containerd-stress \
+                    containerd-shim-runc-v2 \
+                    containerd-shim-runc-v1 \
+                    containerd-shim \
+                    docker-proxy \
+                    sing-box \
+                    cloudflared \
+                    btop \
+                    transmission-daemon \
+                    ttyd \
+                    nano \
+                    jq \
+                    yq \
+                    speedtest-go
+
+define OptimizeTargetBins
+  for bin in $(OPTIMIZE_TARGETS); do \
+    filepath="$(TARGET_DIR)/usr/bin/$$bin"; \
+    if [ -f "$$filepath" ]; then \
+      $(STAGING_DIR_HOST)/bin/sstrip "$$filepath" 2>/dev/null && \
+      $(STAGING_DIR_HOST)/bin/upx --best --lzma "$$filepath" >/dev/null 2>&1; \
+    fi; \
+  done
+endef
+
+TARGET_DIR_POST_INSTALL_HOOKS += $(OptimizeTargetBins)
